### PR TITLE
Migrate Pad Device and All references 

### PR DESCRIPTION
**Pull request by ntarafdar**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/9744)

### Problem description
Completing the migration of TT_Liib to TTNN Pad

### What's changed
Migrated device side code and removed remaining references except for references in tt_lib_ops in sweep tests. That will be ported to ttnn_ops sweep tests by @npetrovic-tenstorrent (CC: @boris-drazic)  

### Checklist
- [x] Post commit CI passes
- [x] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by eyonland — APPROVED** [review]

**Review by cglagovichTT — APPROVED** [review]

**Review by djordje-tt — COMMENTED** [review]
Can you please run t3k pipelines? Demo, frequent, unit and model_perf tests? Just to make sure there is no regression.

File: tests/tt_eager/python_api_testing/sweep_tests/tt_lib_ops.py [hunk 1]
```diff
@@ -2376,10 +2375,10 @@ def pad(
     **kwargs,
 ):
     t0 = setup_tt_tensor(x, device, layout[0], input_mem_config[0], dtype[0])
-    t1 = ttl.tensor.pad(
```
Comment (boris-drazic) [inline tests/tt_eager/python_api_testing/sweep_tests/tt_lib_ops.py:2379]: This file contains ttl operations for sweep tests, so this should stay as ttl or if pad in no longer available in ttl then this is no longer needed and we should remove pad tests. Please let @npetrovic-tenstorrent know if we are removing ttl pad so he can remove tests that use this.

**Comment by tarafdarTT** [comment]
> Can you please run t3k pipelines? Demo, frequent, unit and model_perf tests? Just to make sure there is no regression.

Yes I did run them: 
t3k frequent: https://github.com/tenstorrent/tt-metal/actions/runs/9788704350 (complete, failure same as main)
t3k model perf: https://github.com/tenstorrent/tt-metal/actions/runs/9788706399 (complete, failure same as main)
t3k demo: https://github.com/tenstorrent/tt-metal/actions/runs/9796065498 (in progress)
t3k profiler: https://github.com/tenstorrent/tt-metal/actions/runs/9788710691 (complete, passes)
t3k unit tests: https://github.com/tenstorrent/tt-metal/actions/runs/9788711944 (complete, passes)

**Review by TT-BrianLiu — APPROVED** [review]

**Review by boris-drazic — APPROVED** [review]

**Review by cglagovichTT — APPROVED** [review]
Have you run t3k unit, frequent, and demo tests?

**Comment by tarafdarTT** [comment]
> Have you run t3k unit, frequent, and demo tests?

Yes

**Review by pavlejosipovic — APPROVED** [review]

**Review by pavlepopovic — APPROVED** [review]

**Review by tt-aho — APPROVED** [review]

**Review by djordje-tt — APPROVED** [review]

File: models/demos/resnet/tt/metalResnetBlock50.py [hunk 2]
```diff
@@ -2314,8 +2314,13 @@ def forward(self, x: tt_lib.tensor, write_event=None, op_event=None, final_out_m
...
         else:
-            x = tt_lib.tensor.pad(
-                x, padded_shape, [0, 0, 0, 0], 0, output_mem_config=self.memory_config, use_multicore=True
+            x = ttnn.pad(
+                x,
+                ttnn.Shape(padded_shape),
+                ttnn.Shape([0, 0, 0, 0]),
+                0,
+                output_mem_config=self.memory_config,
+                use_multicore=True,
```
Comment (ayerofieiev-tt) [inline models/demos/resnet/tt/metalResnetBlock50.py:2323]: why do we need `use_multicore`?

File: models/demos/resnet/tt/metalResnetBlock50.py [hunk 3]
```diff
@@ -2360,8 +2365,13 @@ def forward(self, x: tt_lib.tensor, write_event=None, op_event=None, final_out_m
...
             )
         else:
-            x = tt_lib.tensor.pad(
-                x, padded_shape, [0, 0, 0, 0], 0, output_mem_config=self.memory_config, use_multicore=True
+            x = ttnn.pad(
+                x,
+                ttnn.Shape(padded_shape),
+                ttnn.Shape([0, 0, 0, 0]),
+                0,
+                output_mem_config=self.memory_config,
```
Comment (ayerofieiev-tt) [inline models/demos/resnet/tt/metalResnetBlock50.py:2373]: `memory_config`

**Review by ayerofieiev-tt — APPROVED** [review]

File: models/demos/resnet/tt/metalResnetBlock50.py (hunk 2 above)
Comment (tarafdarTT) [reply models/demos/resnet/tt/metalResnetBlock50.py:2323]: initially I thought no models 'use_multicore=False' but many do as that is the default and many models didn't change it to true. 
When I tried changing it (for example in the sweep tests) there were some asserts that broke so for now we will have to leave it in (to keep parity with tt-lib )

**Pull request merged**